plugins/UM3NetworkPrinting/src/Cloud/CloudOutputDeviceManager.py

```diff
@@ -206,19 +228,85 @@ def _onDevicesDiscovered(self, clusters: List[CloudClusterResponse]) -> None:
         max_disp_devices = 3
         if len(new_devices) > max_disp_devices:
             num_hidden = len(new_devices) - max_disp_devices + 1
-            device_name_list = ["- {} ({})".format(device.name, device.printerTypeName) for device in new_devices[0:num_hidden]]
-            device_name_list.append(self.I18N_CATALOG.i18nc("info:hidden list items", "- and {} others", num_hidden))
+            device_name_list = ["<li>{} ({})</li>".format(device.name, device.printerTypeName) for device in new_devices[0:num_hidden]]
+            device_name_list.append(self.I18N_CATALOG.i18nc("info:hidden list items", "<li>... and {} others</li>", num_hidden))
             device_names = "\n".join(device_name_list)
         else:
-            device_names = "\n".join(["- {} ({})".format(device.name, device.printerTypeName) for device in new_devices])
+            device_names = "\n".join(["<li>{} ({})</li>".format(device.name, device.printerTypeName) for device in new_devices])
 
         message_text = self.I18N_CATALOG.i18nc(
             "info:status",
-            "Cloud printers added from your account:\n{}",
+            "Cloud printers added from your account:\n<ul>{}</ul>",
             device_names
         )
         message.setText(message_text)
 
+    def _devicesRemovedFromAccount(self, removed_device_ids: Set[str]) -> None:
+        """
+        Removes the CloudOutputDevice from the received device ids and marks the specific printers as "removed from
+        account". In addition, it generates a message to inform the user about the printers that are no longer linked to
+        his/her account. The message is not generated if all the printers have been previously reported as not linked
+        to the account.
+
+        :param removed_device_ids: Set of device ids, whose CloudOutputDevice needs to be removed
+        :return: None
+        """
+
+        if not CuraApplication.getInstance().getCuraAPI().account.isLoggedIn:
+            return
+
+        # Do not report device ids which have been previously marked as non-linked to the account
+        ignored_device_ids = set()
+        for device_id in removed_device_ids:
+            if not parseBool(self._um_cloud_printers[device_id].getMetaDataEntry(self.META_LINKED_TO_ACCOUNT, "true")):
+                ignored_device_ids.add(device_id)
+        # Keep the reported_device_ids list in a class variable, so that the message button actions can access it and
+        # take the necessary steps to fulfill their purpose.
+        self.reported_device_ids = removed_device_ids - ignored_device_ids
+        if len(self.reported_device_ids) == 0:
```

Comment:
```suggestion
        if not self.reported_device_ids:
```

Don't use len == 0. It's much slower.